packages/react/src/components/RuleBuilder/RuleBuilderEditor.jsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { TableColumnsPropTypes } from '../Table/TablePropTypes';
+
+import Rule, { RuleGroupPropType } from './Rule';
+import RuleBuilderHeader from './RuleBuilderHeader';
+import {
+  addRule,
+  filterRulesById,
+  findRulePathById,
+  generateRuleGroup,
+  getRuleByPath,
+  updateRuleAtPath,
+} from './utils';
+
+const propTypes = {
+  // the rules passed into the component. The RuleBuilder is a controlled component, so
+  // this works the same as passing defaultValue to a controlled input component.
+  defaultRules: RuleGroupPropType,
+  // the columns of the table to be used in the column dropdown.
+  columns: TableColumnsPropTypes.isRequired,
```

Comment:
define the props that needed for this component

Reply:
@stuckless added custom prop types instead of using the table columns. This also includes a render function and per-column operand values.